### PR TITLE
Add autocorrect for Style/SelfAssignment cop

### LINKED ISSUE #1276: Add autocorrect for Style/SelfAssignment cop
References https://github.com/bbatsov/rubocop/issues/743.

### PR DESCRIPTION
**Pull request by lumeet**
Fixes #1276.

### CODE REVIEW

File: lib/rubocop/cop/style/self_assignment.rb [hunk 1]
```diff
@@ -62,6 +62,33 @@ def check_boolean_node(node, rhs, var_name, var_type)
           operator = rhs.loc.operator.source
           add_offense(node, :expression, format(MSG, operator))
         end
+
+        def autocorrect(node)
+          _, rhs = *node
```
Comment (bbatsov) [inline lib/rubocop/cop/style/self_assignment.rb]: I generally avoid using a plain `_` as it decreases the readability of the code a bit. You know what you're destructuring, but another reader of code might not know that.

File: lib/rubocop/cop/style/self_assignment.rb [hunk 2]
```diff
@@ -62,6 +62,33 @@ def check_boolean_node(node, rhs, var_name, var_type)
...
+
+          if rhs.type == :send
+            autocorrect_send_node(node, rhs)
+          elsif [:and, :or].include?(rhs.type)
+            autocorrect_boolean_node(node, rhs)
+          end
+        end
+
+        def autocorrect_send_node(node, rhs)
+          _, operator, rest = *rhs
```
Comment (bbatsov) [inline lib/rubocop/cop/style/self_assignment.rb]: I'd suggest using some more descriptive name instead of the generic `rest`.

**Comment by bbatsov** [comment]
Apart from my small remarks the code looks good! :+1:

**Comment by lumeet** [comment]
Thanks, that was constructive. I'm just wondering why I overlooked the naming in the check methods in the first place. Fixed.

**Comment by bbatsov** [comment]
Thanks! Keep those improvements coming! :-)

**Pull request merged**